src/util/AvaraGL.cpp

```diff
@@ -192,6 +209,8 @@ void AvaraGLSetAmbient(float ambient, ARGBColor color) {
 void ActivateLights(float active) {
     glUseProgram(gProgram);
     glUniform1f(lights_activeLoc, active);
+    glUseProgram(hudProgram);
```

Comment:
This might be causing the lighting issues you were talking about. This is using the new shader instead of the old one for for most of AvaraGLDrawPolygons().